src/client/store/geo/mosaic/actions/setOption.ts

```diff
@@ -2,9 +2,12 @@ import { createAction } from '@reduxjs/toolkit'
 
 import { MosaicOptions } from 'meta/geo'
 
-type Params<K extends keyof MosaicOptions = keyof MosaicOptions> = {
-  key: K
-  value: MosaicOptions[K]
-}
+// Property key to set, and its value type
+type Params = {
+  [K in keyof MosaicOptions]: {
+    key: K
+    value: MosaicOptions[K]
+  }
+}[keyof MosaicOptions]
```

Comment:
This part seems duplicated, not sure if we can introduce generic or ts util to do this?